File: nameAbbreviater.go
package main

import (
	"fmt"
	"strings"
)

func abbreviateName(fullName string, fNameStatement bool, mNameStatement bool, sNameStatement bool) {
	slicedFullName := strings.Split(fullName, " ")
    midNames := slicedFullName[:len(slicedFullName)-1]
	midNames = midNames[1:]
	output := ""
	dot := "."
	space := " "

	if fNameStatement == true {
		output += string(slicedFullName[0][0]) + dot + space
	}

	if len(midNames) > 1 {
		if mNameStatement == true {
			for i, s := range midNames {
				output += string(s[i-i]) + "." + " "
			}
		}
	}

	if sNameStatement == true {
		output += string(slicedFullName[len(slicedFullName)-1][0]) + dot + space
	}

	fmt.Println(output)
}

func main() {
	abbreviateName("Mustafa Samir Veli Dili", true, true, true)
}

File: nameAbbreviater.py
def abbreviateName(fullName, fNameStatement, mNameStatement, sNameStatement):
    output = ""
    nameLenLimit = 3
    sFullName = fullName.split()
    fName = sFullName[0]
    sName = sFullName[-1]
    if fNameStatement is True:
        output += setOutputType(fNameStatement, fName[0])
    else:
        output += setOutputType(fNameStatement, fName[0])

    # get all elements between the first and the last elements
    midNames = sFullName[1:-1]
    if mNameStatement is True:
        if len(sFullName) >= nameLenLimit:
            if len(midNames) >= 0:
                for midName in midNames:
                    output += setOutputType(mNameStatement, midName[0])
    else:
        for midName in midNames:
            print(output)
            output += setOutputType(fNameStatement, midName)

    if sNameStatement is True:
        output += setOutputType(sNameStatement, sName[0])
    else:
        output += setOutputType(fNameStatement, sName)
    print(output)


def setOutputType(short, name):
    if short is False:
        return name + " "
    else:
        return name[0] + "." + " "


fNameStatement = True
mNameStatement = True
sNameStatement = False
abbreviateName("Mustafa Kemal Demir", fNameStatement,
               mNameStatement, sNameStatement)
